mds/provider_poller/poller.py

```diff
@@ -486,10 +609,21 @@ def _create_event_record(status_change):
     else:  # Spec violation!
         point = None
 
-    first_saved_at = None
-    # "Aggregation" layers store a recorded field, and we want to keep the same value
-    if "recorded" in status_change and status_change["recorded"]:
-        first_saved_at = utils.from_mds_timestamp(status_change["recorded"])
+    publication_time = None
+    if status_change.get("publication_time"):
+        publication_time = utils.from_mds_timestamp(status_change["publication_time"])
+    # "Aggregation" providers store a recorded field, and we want to keep the same value
+    # until we get the publication_time everywhere
+    # Vendor only, 0.3 only, will disappear
+    recorded = None
+    if status_change.get("recorded"):
+        recorded = utils.from_mds_timestamp(status_change["recorded"])
+    if publication_time and recorded:
+        difference = abs(publication_time - recorded)
+        if difference > datetime.timedelta(minutes=10):
+            logger.warning("publication_time and recorded differ by %s", difference)
+    if not publication_time and recorded:
```

Comment:
You can place by setting `publication_time=publication_time or recorded` in `EventRecord` creation